source/game.gamescriptexpression.bmx

```diff
@@ -611,7 +611,7 @@ Function SEFN_role:SToken(params:STokenGroup Var, context:SScriptExpressionConte
 	Local role:TProgrammeRole
 	Local token:SToken = params.GetToken(1)
 	Local GUID:String = token.value
-	Local ID:Long = token.valueLong
+	Local ID:Long = token.GetValueLong()
```

Comment:
Bitte beachte, dass "Casts" von "string zu long" komische Ergebnisse liefern koennten!

```
Print Long("12-1234")
'-> 12

Print Long("12a-a1234")
'-> 12
```
Das heisst, eine nicht gefundene GUID, die aber zufaellig mit einer Zahl beginnt und als Long somit eine Zahl<>0 wird... koennte (zufaellig) eine valide ID werden.

Reply:
Mit neueren Blitzmaxversionen (bzw blitz.mod) bekommen wir "meinstring.ToDoubleEx()/ToLongEx()..." und bekommen dort die Dinge schneller, fehlerfreier und konfigurierbar.
Allerdings erzeugt es ebenfalls "12" (nur dann mit der Info, dass ab Char-Index 2 "non-long characters" gefunden wurden).

```
SuperStrict
Framework Brl.StandardIO

Print Long("12a-a1234")

Local l:Long
Print "result: " + "12a-a1234".ToLongEx(l)
Print "l  " + l
```
Ausgabe:
```
12
result: 2
l  12
```

doc:
```
// extracts a Long from a string, from the range startPos to endPos
// endPos of -1 means the end of the string
// returns 0 if the string is not a valid Long, or the position of the first character after the Long otherwise
```